setup.py

```diff
@@ -30,6 +30,8 @@ def read_req_file(req_type):
     extras_require={
         "dev": read_req_file("dev"),
         "testing": read_req_file("testing"),
+        "celery-manager": read_req_file("celery-manager"),
+        "diskcache-manager": read_req_file("diskcache-manager"),
```

Comment:
This makes sense, thanks @anders-kiaer!

Let's include `ci` as an option here too. It's less important now that dcc, html, and table are combined into this repo, but we still like to use the same CI setup in other non-core component repos. And perhaps we can shorten the new ones by dropping `-manager`, just `celery` and `diskcache`?

Reply:
Is there a convention for how to document available extras? We can of course add this to the docs site (not sure we want to go into all this detail in https://dash.plotly.com/installation but elsewhere on that site we can) but what about documentation within the package itself?

Perhaps a comment at the top of each `requires-*` file describing its purpose?

Reply:
> And perhaps we can shorten the new ones by dropping `-manager`, just `celery` and `diskcache`?

Agree, I can shorten name the groups by dropping `-manager`. As well as make a `ci` option. 👍 

> Is there a convention for how to document available extras? We can of course add this to the docs site (not sure we want to go into all this detail in https://dash.plotly.com/installation but elsewhere on that site we can) but what about documentation within the package itself? Perhaps a comment at the top of each requires-* file describing its purpose?

The closest thing to a convention that I've seen is that they are documented along with the extra features depending on these extra dependencies (so that goes along your suggestion with not documenting it on https://dash.plotly.com/installation, but instead where the celery/diskcache managers are documented). A comment in each `requires-*` file feels natural, and these comments could easily be moved along into e.g. a [`pyproject.toml` optional section](https://www.python.org/dev/peps/pep-0621/#dependencies-optional-dependencies) or `setup.cfg` if doing that change at some point.